File: app/src/main/java/com/example/ridvan/doctorandpatientfirebase/HospitalAdapter.java
package com.example.ridvan.doctorandpatientfirebase;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import java.util.ArrayList;

public class HospitalAdapter extends RecyclerView.Adapter<HospitalAdapter.MyViewHolder> {
    ArrayList<Hospital> mDataList;
    LayoutInflater inflater;
    public HospitalAdapter(Context context, ArrayList<Hospital> data) {
        //inflater= (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater = LayoutInflater.from(context);
        this.mDataList = data;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = inflater.inflate(R.layout.hospital_list, parent, false);
        MyViewHolder holder = new MyViewHolder(v);
        return holder;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        Hospital tiklanilanHospital = mDataList.get(position);
        holder.setData(tiklanilanHospital, position);
    }

    @Override
    public int getItemCount() {
        return mDataList.size();
    }
    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView textViewHospitalName,textViewHospitalCity;
        public MyViewHolder(View itemView) {
            super(itemView);
            textViewHospitalName = (TextView) itemView.findViewById(R.id.textViewHospitalName);
            textViewHospitalCity = (TextView) itemView.findViewById(R.id.textViewHospitalCity);
        }
        public void setData(Hospital tiklanilanHospital, int position) {
            this.textViewHospitalName.setText( tiklanilanHospital.getHospital_city() );
            this.textViewHospitalCity.setText(tiklanilanHospital.getHospital_name());
        }
    }
}

File: app/src/main/java/com/example/ridvan/doctorandpatientfirebase/HospitalAdapter.kt
package com.example.ridvan.doctorandpatientfirebase

import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.hospital_list.view.*
import java.util.ArrayList
    /**
     * Kendi Oluşturacağımız view Holderın Adını Yazdık -> HospitalsViewHolder
     */
class HospitalAdapter(allHospitals: ArrayList<HospitalDataModel>) : RecyclerView.Adapter<HospitalAdapter.HospitalsViewHolder>() {
    var hospitalAll=allHospitals
    override fun getItemCount(): Int {
        //Adapter iik çalıştığı zaman buraya bakar
        return hospitalAll.size
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HospitalsViewHolder {
        //viewHolder burada oluşturulacak (inflater işlemi)
        var inflater= LayoutInflater.from(parent?.context)
        var oneLineHospital=inflater.inflate(R.layout.hospital_list,parent,false)

        return HospitalsViewHolder(oneLineHospital)
    }
    override fun onBindViewHolder(holder: HospitalsViewHolder, position: Int) {
        //benim oluşturduğum sınıfın nesnesini geriğe dönecek

        holder?.textViewHospitalName?.text=hospitalAll.get(position).hospital_name
        holder?.textViewHospitalCity?.text=hospitalAll.get(position).hospital_city
    }

    class HospitalsViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {
        //içindeki elemanlara buradan erişeceğim
        var oneLineHospital=itemView as CardView
        var textViewHospitalName=oneLineHospital.dataDate
        var textViewHospitalCity=oneLineHospital.textViewHospitalCity

    }
}